eng/pipelines/templates/stages/workload-public-build.yml

```diff
@@ -0,0 +1,32 @@
+parameters:
+  sourceBranch: self
+  engBranch: eng
+
+stages:
+- stage: Build_${{ parameters.sourceBranch }}
+  displayName: Build ${{ parameters.sourceBranch }}
+  dependsOn: []
+  jobs:
+  - template: /eng/common/templates/job/job.yml@${{ parameters.sourceBranch }}
+    parameters:
+      name: buildRepo
+      displayName: Build Repo
+      pool:
+        name: $(DncEngPublicBuildPool)
+        demands: ImageOverride -equals windows.vs2022.amd64.open
```

Comment:
TODO (for myself): Consider centralizing the declaration of the images. Then, pass in as a parameter here.